src/App.tsx

```diff
@@ -15,49 +17,77 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+function getPreparedGoods(goods: string[],
+  sortBy: string,
+  isReversed: boolean) {
+  const preparedGoods = [...goods];
+
+  if (sortBy === 'alphabet') {
+    preparedGoods.sort((a, b) => a.localeCompare(b));
+  }
+
+  if (sortBy === 'length') {
+    preparedGoods.sort((a, b) => a.length - b.length);
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortBy, isReversed);
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
+          onClick={() => setSortBy('alphabet')}
           type="button"
-          className="button is-info is-light"
+          className={`button is-info ${sortBy === 'alphabet' ? '' : 'is-light'}`}
```

Comment:
use classnames utility to add class dependent from condition

import cn from 'classnames';

Reply:
https://www.npmjs.com/package/classnames